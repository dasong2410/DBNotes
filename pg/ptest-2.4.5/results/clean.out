CREATE TABLE clean_cases (origin varchar, caseno numeric,
                          orig geometry,
                          valid geometry);
select sn_create_distributed_table('clean_cases', 'origin', 'none');
ERROR:  type "public.geometry" does not exist
CONTEXT:  while executing command on localhost:57637
SQL statement "SELECT pg_catalog.create_distributed_table('clean_cases','origin','hash','none')"
PL/pgSQL function sn_create_distributed_table(regclass,text,text) line 58 at EXECUTE
COPY clean_cases FROM stdin;
-- PG.1 : polygon with single ring with single point in it
--        to be converted to a POINT
--
SELECT origin,caseno,
       st_equals(st_collectionextract(st_makevalid(orig),1), st_collectionextract(valid,1)) AND
       st_equals(st_collectionextract(st_makevalid(orig),2), st_collectionextract(valid,2)) AND
       st_equals(st_collectionextract(st_makevalid(orig),3), st_collectionextract(valid,3)),
       st_isvalid(st_makevalid(orig)), -- paranoia
       (st_isvaliddetail(orig)).valid
  FROM clean_cases;
 origin | caseno | ?column? | st_isvalid | valid 
--------+--------+----------+------------+-------
 RT     |      1 | t        | t          | f
 RT     |      2 | t        | t          | f
 RT     |      3 | t        | t          | f
 RT     |      4 | t        | t          | f
 RT     |      5 | t        | t          | f
 RT     |      6 | t        | t          | f
 RT     |      7 | t        | t          | f
 RT     |    7.1 | t        | t          | f
 RT     |      8 | t        | t          | f
 RT     |      9 | t        | t          | f
 RT     |    9.1 | t        | t          | f
 RT     |     10 | t        | t          | t
 RT     |     11 | t        | t          | f
 RT     |     12 | t        | t          | f
 RT     |   13.1 | t        | t          | t
 RT     |   13.2 | t        | t          | f
 RT     |     14 | t        | t          | f
 RT     |     15 | t        | t          | f
 RT     |   16.1 | t        | t          | f
 RT     |   16.2 | t        | t          | f
 RT     |   16.3 | t        | t          | f
 RT     |   16.4 | t        | t          | f
 RT     |   17.1 | t        | t          | f
 PG     |      1 | t        | t          | f
 PG     |      2 | t        | t          | f
 PG     |      3 | t        | t          | f
 PG     |      4 | t        | t          | f
 PG     |      5 | t        | t          | f
(28 rows)

SELECT '#1719.1', ST_AsEWKT(ST_MakeValid('POINT(0 0)'));
 ?column? | st_asewkt  
----------+------------
 #1719.1  | POINT(0 0)
(1 row)

SELECT '#1719.2', ST_AsEWKT(ST_MakeValid('GEOMETRYCOLLECTION(POINT(0 0),MULTIPOINT(3 4,5 2),LINESTRING(4 4, 4 4),POLYGON((0 0,10 10,0 10,10 0,0 0)))'));
 ?column? |                                                        st_asewkt                                                        
----------+-------------------------------------------------------------------------------------------------------------------------
 #1719.2  | GEOMETRYCOLLECTION(POINT(0 0),MULTIPOINT(3 4,5 2),POINT(4 4),MULTIPOLYGON(((0 0,5 5,10 0,0 0)),((5 5,0 10,10 10,5 5))))
(1 row)

SELECT '#1719.3', ST_AsEWKT(ST_MakeValid('MULTIPOINT(3 4,5 2)'));
 ?column? |      st_asewkt      
----------+---------------------
 #1719.3  | MULTIPOINT(3 4,5 2)
(1 row)

DROP TABLE clean_cases;
