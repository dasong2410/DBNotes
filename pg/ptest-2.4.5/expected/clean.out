CREATE TABLE clean_cases (origin varchar, caseno numeric,
                          orig geometry,
                          valid geometry);
select sn_create_distributed_table('clean_cases', 'origin', 'none');
 sn_create_distributed_table 
-----------------------------
 
(1 row)

COPY clean_cases FROM stdin;
ERROR:  Polygon must have closed rings
-- PG.1 : polygon with single ring with single point in it
--        to be converted to a POINT
--
SELECT origin,caseno,
       st_equals(st_collectionextract(st_makevalid(orig),1), st_collectionextract(valid,1)) AND
       st_equals(st_collectionextract(st_makevalid(orig),2), st_collectionextract(valid,2)) AND
       st_equals(st_collectionextract(st_makevalid(orig),3), st_collectionextract(valid,3)),
       st_isvalid(st_makevalid(orig)), -- paranoia
       (st_isvaliddetail(orig)).valid
  FROM clean_cases;
 origin | caseno | ?column? | st_isvalid | valid 
--------+--------+----------+------------+-------
(0 rows)

SELECT '#1719.1', ST_AsEWKT(ST_MakeValid('POINT(0 0)'));
 ?column? | st_asewkt  
----------+------------
 #1719.1  | POINT(0 0)
(1 row)

SELECT '#1719.2', ST_AsEWKT(ST_MakeValid('GEOMETRYCOLLECTION(POINT(0 0),MULTIPOINT(3 4,5 2),LINESTRING(4 4, 4 4),POLYGON((0 0,10 10,0 10,10 0,0 0)))'));
 ?column? |                                                        st_asewkt                                                        
----------+-------------------------------------------------------------------------------------------------------------------------
 #1719.2  | GEOMETRYCOLLECTION(POINT(0 0),MULTIPOINT(3 4,5 2),POINT(4 4),MULTIPOLYGON(((0 0,5 5,10 0,0 0)),((5 5,0 10,10 10,5 5))))
(1 row)

SELECT '#1719.3', ST_AsEWKT(ST_MakeValid('MULTIPOINT(3 4,5 2)'));
 ?column? |      st_asewkt      
----------+---------------------
 #1719.3  | MULTIPOINT(3 4,5 2)
(1 row)

DROP TABLE clean_cases;
